src/main/java/com/kodarit/stars/toolbox/ImageResizer.java

```diff
@@ -0,0 +1,104 @@
+package com.kodarit.stars.toolbox;
+
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.imageio.ImageIO;
+import java.awt.Graphics2D;
+import java.awt.RenderingHints;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Base64;
+
+@Service
+public class ImageResizer {
+
+    /**
+     * Resizes an image to the specified width and height
+     *
+     * @param imageFile the image file to resize
+     * @param width the target width
+     * @param height the target height
+     * @return Base64 encoded string of the resized image
+     * @throws IOException if image processing fails
+     */
+    public String resizeImage(MultipartFile imageFile, int width, int height) throws IOException {
+        // Read the original image
+        BufferedImage originalImage = ImageIO.read(imageFile.getInputStream());
+        if (originalImage == null) {
+            throw new IOException("Could not read image format");
+        }
+        
+        // Use BufferedImage.TYPE_INT_ARGB to preserve transparency for PNGs
+        int imageType = originalImage.getTransparency() == BufferedImage.OPAQUE ? 
+                BufferedImage.TYPE_INT_RGB : BufferedImage.TYPE_INT_ARGB;
+        
+        // Create a new BufferedImage for the resized image
+        BufferedImage resizedImage = new BufferedImage(width, height, imageType);
+
+        // Draw the original image scaled to the new dimensions with improved quality
+        Graphics2D g = resizedImage.createGraphics();
```

Comment:
Single letter variable names are kind of bad, even if it's pretty obvious here. We don't need to save a few characters.